tests/test_pickling.cpp

```diff
@@ -60,6 +63,47 @@ void wrap(py::module m) {
 
 } // namespace exercise_trampoline
 
+namespace exercise_getinitargs_getstate_setstate {
+
+// Mechanism similar to what was established with Boost.Python:
```

Comment:
You may want to actually describe the protocol here rather than just linking to the origin. My stab at this based on your PR description is that:

Test binding __set_state__ to a non-constructor method, via `__nonconstructor_set_state__`.

Reply:
Please take another look. I changed the comment here significantly.

For the description of the protocol I'm pointing to https://www.boost.org/doc/libs/1_31_0/libs/python/doc/v2/pickle.html now (the page is actually from 2002). The main point is that `__setstate__[non-constructor]` is to support something that is long established.